lib/matplotlib/tests/test_sphinxext.py

```diff
@@ -122,6 +122,50 @@ def test_plot_html_show_source_link(tmpdir):
     assert "index-1.py" not in [p.name for p in html_dir2.iterdir()]
 
 
+@pytest.mark.parametrize('plot_html_show_source_link', [0, 1])
+def test_show_source_link_true(tmpdir, plot_html_show_source_link):
```

Comment:
```suggestion
def test_show_source_link_true(tmp_path, plot_html_show_source_link):
```

[tmp_path](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html) is the modern replacement for `tmpdir` and provides a `Path` instance immediately.

Reply:
Done.